modules/pubwiseAnalyticsAdapter.js

```diff
@@ -54,72 +73,268 @@ function enrichWithUTM(dataBag) {
+    }
+
+    if (configOptions.custom.c_slot2) {
+      dataBag['c_slot2'] = configOptions.custom.c_slot2;
+    }
+
+    if (configOptions.custom.c_slot3) {
+      dataBag['c_slot3'] = configOptions.custom.c_slot3;
+    }
+
+    if (configOptions.custom.c_slot4) {
+      dataBag['c_slot4'] = configOptions.custom.c_slot4;
+    }
+  }
+
+  return dataBag;
+}
+
+function setNamespace(itemText) {
+  return pwNamespace.concat('_' + itemText);
+}
+
+function localStorageSessTimeoutName() {
+  return setNamespace(sessTimeoutName);
+}
+
+function localStorageSessName() {
+  return setNamespace(sessName);
+}
+
+function extendUserSessionTimeout() {
+  localStorage.setItem(localStorageSessTimeoutName(), Date.now().toString());
+}
+
+function userSessionID() {
+  return localStorage.getItem(localStorageSessName()) ? localStorage.getItem(localStorageSessName()) : '';
+}
+
+function sessionExpired() {
+  let sessLastTime = localStorage.getItem(localStorageSessTimeoutName());
+  return (Date.now() - parseInt(sessLastTime)) > sessTimeout;
+}
+
+function flushEvents() {
+  if (pwEvents.length > 0) {
+    let dataBag = {metaData: metaData, eventList: pwEvents.splice(0)}; // put all the events together with the metadata and send
+    ajax(configOptions.endpoint, (result) => pwInfo(`Result`, result), JSON.stringify(dataBag));
+  }
+}
+
+function isIngestedEvent(eventType) {
+  const ingested = [
+    CONSTANTS.EVENTS.AUCTION_INIT,
+    CONSTANTS.EVENTS.BID_REQUESTED,
+    CONSTANTS.EVENTS.BID_RESPONSE,
+    CONSTANTS.EVENTS.BID_WON,
+    CONSTANTS.EVENTS.BID_TIMEOUT,
+    CONSTANTS.EVENTS.AD_RENDER_FAILED,
+    CONSTANTS.EVENTS.TCF2_ENFORCEMENT
+  ];
+  return ingested.indexOf(eventType) !== -1;
+}
+
+function markEnabled() {
+  pwInfo(`Enabled`, configOptions);
+  pwAnalyticsEnabled = true;
+  setInterval(flushEvents, 100);
+}
+
+function pwInfo(info, context) {
+  utils.logInfo(`${analyticsName} ` + info, context);
+}
+
+/*
+  // unused currently
+  function filterNoBid(data) {
+    let newNoBidData = {};
 
-  // put the typical items in the data bag
-  let dataBag = {
-    eventType: eventType,
-    args: data,
-    target_site: configOptions.site,
-    pubwiseSchema: pubwiseSchema,
-    debug: configOptions.debug ? 1 : 0,
-  };
+    newNoBidData.auctionId = data.auctionId;
+    newNoBidData.bidId = data.bidId;
+    newNoBidData.bidderRequestId = data.bidderRequestId;
+    newNoBidData.transactionId = data.transactionId;
 
-  dataBag = enrichWithMetrics(dataBag);
-  // for certain events, track additional info
-  if (eventType == CONSTANTS.EVENTS.AUCTION_INIT) {
-    dataBag = enrichWithUTM(dataBag);
+    return newNoBidData;
   }
+*/
 
-  ajax(configOptions.endpoint, (result) => utils.logInfo(`${analyticsName}Result`, result), JSON.stringify(dataBag));
+function filterBidResponse(data) {
+  let modified = Object.assign({}, data);
+  // clean up some properties we don't track in public version
+  if (typeof modified.ad !== 'undefined') {
```

Comment:
Alternative syntax to be considered: 
```
modified.ad = modified.ad || ''
``` 
warn that if `modified.ad` value can be `0`,  it will become `''`.

Reply:
We don't rely on truthiness for data not under our control. Paranoid? Maybe? Unnecessary? Maybe. We write in 6 different languages and would like to reliably get the meaning we want. Our desire is to determine if the values is, but often is not, defined and underlying behavior in various languages around what happens when the value is defined but "trueish" is variable. So, we make explicit statements when the value comes from somewhere else.